### PR TITLE
Use pip+venv in get-started/installation.md

### PR DESCRIPTION
**Pull request by sfc-gh-kbregula**
## 🧠 Description of Changes

Spec: https://www.notion.so/snowflake-corp/Migrate-to-pip-for-dependencies-management-and-drop-pipenv-2ab55d45ed2a4ae3be500c70cd2e73b9

We migrated from `pipenv` to `pip` for dependency management, so I'm updating the documentation to recommend pip+venv for virtual environment management.

I'm opening this as a draft because I'd like to add documentation on the constraints files that advanced users might want to use for installation, but the constraints files aren't finished yet.
https://www.notion.so/snowflake-corp/Constraint-files-Spec-deded93f49fc43ebb01624124bd8b1bc

**Contribution License Agreement**

By submitting this pull request you agree that all contributions to this project are made under the Apache 2.0 license.

### CODE REVIEW

**Comment by sfc-gh-kbregula** [comment]
@snehankekre The last PR for constraints is waiting to be merged -  https://github.com/streamlit/streamlit/pull/6495 . Where should we add documentation on this?

**Comment by snehankekre** [comment]
Am I right in assuming that these constraint files are relevant to developers of the Streamlit library and not users? If so, documentation for it should be in the [Contribution wiki](https://github.com/streamlit/streamlit/wiki/Contributing), not the docs.

**Comment by sfc-gh-kbregula** [comment]
@snehankekre constraints files will be used by end users who encounter problems with dependencies or value stability over new versions of the library will be able to use “known-to-be-working” dependencies.

Here is the spec about constraints files: https://www.notion.so/snowflake-corp/Constraint-files-Spec-deded93f49fc43ebb01624124bd8b1bc
Here are similar docs in Apache Airflow project: https://airflow.apache.org/docs/apache-airflow/stable/installation/installing-from-pypi.html
We also had similar documentation for Snowflake, but it was removed. 
https://github.com/snowflakedb/snowflake-prod-docs/pull/2759/files#r1116327691
Here is the last documentation that described the requirements files. They used requirements files as they wanted to enforce the installation of all dependencies.
https://github.com/snowflakedb/snowflake-prod-docs/blob/021d056d8ea72b68df4ec87c67cd498b18c76da5/sphinx/source/user-guide/python-connector-install.rst

**Comment by snehankekre** [comment]
Thanks for all the context! Docs for constraint files should then go into the same installation docs as this PR, similar to Apache Airflow. I can review in detail on Monday if that works!

**Comment by snehankekre** [comment]
>  but the constraints files aren't finished yet.

@sfc-gh-kbregula are the constraint files launched/finished? If so, I think you could add the docs on it in the same installation doc. Similar to the Apache Airflow docs, we can make `Constraint files` a high level h2 heading, right?

File: content/library/get-started/installation.md [hunk 1]
```diff
@@ -119,26 +113,20 @@ xcode-select --install
...
-2. Create a new Pipenv environment in that folder and activate that environment:
+2. Create a new virtual environment in that folder and activate that environment:
 
    ```sh
-   pipenv shell
+   python -m venv venv
    ```
 
-   When you run the command above, a file called `Pipfile` will appear in `myprojects/`. This file is where your Pipenv environment and its dependencies are declared.
+   When you run the command above, a directory called `venv` will appear in `myprojects/`. This directory is where your virtual environment and its dependencies are install.
```
Comment (snehankekre) [inline content/library/get-started/installation.md:122]:
```suggestion
   When you run the command above, a directory called `venv` will appear in `myprojects/`. This directory is where your virtual environment and its dependencies are installed.
```

**Review by snehankekre — APPROVED** [review]
Approved pending Constraint files documentation 👍

**Comment by MathCatsAnd** [comment]
Does line 80 also need to be updated since it still mentions pipenv?

https://github.com/streamlit/docs/blob/ece40476ecbc9e7b46d812cde3f36268b6c39db6/content/library/get-started/installation.md?plain=1#L80

**Comment by sfc-gh-kbregula** [comment]
@snehankekre Updated.

**Comment by snehankekre** [comment]
@sfc-gh-kbregula Can we merge before constraint files are documented?

**Comment by sfc-gh-kbregula** [comment]
@snehankekre Yes. I think we can do that.

**Comment by snehankekre** [comment]
On 55a307, I just clarified that the environment manager is venv and package manager is pip.

**Pull request merged**